src/Zend/Locale/Format.php

```diff
@@ -1199,34 +1199,34 @@ public static function checkDateFormat($date, array $options = array())
         $options = self::_checkOptions($options) + self::$_options;
 
         // day expected but not parsed
-        if ((iconv_strpos($options['date_format'], 'd', 0, 'UTF-8') !== false) and (!isset($date['day']) or ($date['day'] === ''))) {
```

Comment:
A couple of tests that hit this were failing in PHP 8 but not in PHP 7. I believe the tests were incorrect based on the other tests in the suite. Digging a bit, PHP 7 was returning `false` for the missing date segments, but the code was only checking if it was set, or if it was an empty string, so adding the false check.